### PR TITLE
[Rerun RP Pytest Plugin Improvement] Rerun Upgrade Tests

### PR DESCRIPTION
**Pull request by jyejare**
Until now/today, The Rerun RP plugin was helpful to rerun regular tier tests, but with this improvement, one can also rerun upgrade failed/skipped tests on an upgraded setup.

The usage is simple and just adds one more option `upgrades-rerun` to this plugin.

Usage:
```
# pytest tests/foreman/ --only-failed --upgrades-rerun
```

### CODE REVIEW

File: pytest_plugins/rerun_rp/rerun_rp.py [hunk 1]
```diff
@@ -83,25 +83,28 @@ def pytest_addoption(parser):
...
-            Usage: --user [value]
+        Usage: --user [value]
 
-            Value: [ Report_portal_username ]
-
-            Defect_types:
-                Collect tests of specific user from Report Portal
-        '''
+        Value: [ Report_portal_username ]
+    '''
```
Comment (jyejare) [inline pytest_plugins/rerun_rp/rerun_rp.py:100]: This change is not related to the upgrade rerun improvement. This is just a nitpick. Just FYI to avoid the confusion.

File: pytest_plugins/rerun_rp/rerun_rp.py [hunk 2]
```diff
@@ -83,25 +83,28 @@ def pytest_addoption(parser):
...
     parser.addoption("--user", nargs='?', help=help_text)
+    help_text = '''
+        Rerun Upgrade launch tests instead of Regular Tier launch from Report Portal,
+
+        Usage: --upgrade-rerun
```
Comment (jameerpathan111) [inline pytest_plugins/rerun_rp/rerun_rp.py:105]:
```suggestion
        Usage: --upgrades-rerun
```
typo?

File: pytest_plugins/rerun_rp/rerun_rp.py [hunk 3]
```diff
@@ -83,25 +83,28 @@ def pytest_addoption(parser):
...
+    '''
+    parser.addoption("--upgrades-rerun", action='store_true', default=False, help=help_text)
```
Comment (devendra104) [inline pytest_plugins/rerun_rp/rerun_rp.py:107]: "upgrade-tests-rerun" is more appropriate than "upgrades-rerun"

Comment (jyejare) [reply pytest_plugins/rerun_rp/rerun_rp.py:107]: @devendra104 As pytest and rerun plugin is used for running tests and this plugin specially plays with test collection, so I think I need not to explicitly add `test` word there.

So I am good with whats it now.

**Comment by jyejare** [comment]
@jameerpathan111 @devendra104 Can you merge this! I have addressed your comments !

**Review by devendra104 — APPROVED** [review]
ACK

**Review by jameerpathan111 — APPROVED** [review]
ACK

**Comment by jyejare** [comment]
@SatelliteQE/airgun-tier-2-reviewers VERY VERY LONG pending PR. Please review :)

File: pytest_plugins/rerun_rp/rerun_rp.py [hunk 4]
```diff
@@ -113,13 +116,20 @@ def pytest_collection_modifyitems(items, config):
...
         return
     rp = ReportPortal()
     version = settings.server.version
     sat_version = f'{version.base_version}.{version.epoch}'
     LOGGER.info(f'Fetching Report Portal launches for target Satellite version: {sat_version}')
-    launch = next(iter(rp.launches(sat_version=sat_version).values()))
+    launch = next(
+        iter(
+            rp.launches(
+                sat_version=sat_version, launch_type='upgrades' if upgrades_rerun else 'satellite6'
```
Comment (mshriver) [inline pytest_plugins/rerun_rp/rerun_rp.py:129]: This hardcoding (and use of) `'satellite6'` is something we need to move away from. The launch type should be pulled from report portal configuration, or directly passable.

**Review by mshriver — APPROVED** [review]

File: pytest_plugins/rerun_rp/rerun_rp.py (hunk 4 above)
Comment (jyejare) [reply pytest_plugins/rerun_rp/rerun_rp.py:129]: @mshriver by configuration you mean from current robottelo.properties (future dynaconf) ...  that should be doable and need since we are moving to DynaConf.

Comment (jyejare) [reply pytest_plugins/rerun_rp/rerun_rp.py:129]: We have that already available in RP, but thats basically for Report Portal Module. For plugin I just wanted to make it easier by `--upgrades_rerun` option.
